codeSnippets/snippets/custom-plugin-authorization/src/main/kotlin/com/example/Application.kt

```diff
@@ -0,0 +1,58 @@
+package com.example
+
+import com.example.plugins.*
+import io.ktor.server.application.*
+import io.ktor.server.auth.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+
+fun main(args: Array<String>): Unit = io.ktor.server.netty.EngineMain.main(args)
+
+fun getUserRole(userName: String?): String {
+    val userRole = when (userName) {
+        "jetbrains" -> "user"
+        "admin" -> "admin"
+        else -> ""
+    }
+    return userRole
+}
+
+fun Application.module() {
+    install(Authentication) {
+        basic("auth-basic") {
+            realm = "Access to the '/' path"
+            validate { credentials ->
+                if (credentials.name == "jetbrains" ||
```

Comment:
I will suggest to not rely here on a priority between `&&` and `||` and use brackets